### ISSUE TITLE
Incorrect new version email

### ISSUE THREAD

**Issue by jamesjer** [fedora-infra/anitya#1167]
I just got an email informing me that:

A new version of "pynac" has been detected: "0.9.7" newer than "0.4.2", packaged as "pynac"

This is wrong twice.  First, it is wrong because the version of pynac in Fedora is 0.7.27, not 0.4.2.  It has never been 0.4.2.  The pynac package went from version 0.3.9 to version 0.6.7 in 2016.

Second, it is wrong because the latest version of pynac is 0.7.29, not 0.9.7.  I looked at the release-monitoring.org page for pynac, and found the problem.  Somebody or something changed the new version URL to point to the pypi.org page for pynac.  Only that's a completely different project.  It has no relationship at all with the Fedora package named pynac.  So now the version history is poisoned with incorrect version numbers.  I'll try to straighten that out, but if you can tell who or what made that change, please tell him/her/it to be more careful in the future.

**Comment by rwmjones** [comment]
Similar thing happened with ocamlbuild, see this thread:

https://lists.fedoraproject.org/archives/list/devel@lists.fedoraproject.org/thread/UVAQP4DEOO7PAIKWMFORV4MSZDFNJIFU/

**Comment by Zlopez** [comment]
@jamesjer The Anitya is opened for everyone to edit the projects, so if somebody changed this they probably thought it's a right project. There isn't much I can do in Anitya to prevent this.

@rwmjones Not sure where the version comes from, but this version is really there https://github.com/ocaml/ocamlbuild/releases/tag/4.02.3 You can edit the project in Anitya to ignore those versions and I can delete them, but this looks like the project is tagging wrong things. Maybe using releases instead tags could help. In this case it will only retrieve tags, which were marked as GitHub release in project.

I'm closing this, because it's not much I can do on the Anitya side.

**Comment by rwmjones** [comment]
This doesn't address the problem at all.  The email said the latest version was 0.10.0 but it's 0.14.0.

**Comment by Zlopez** [comment]
the-new-hotness is taking this information from mdapi and this really says 0.14.0 https://mdapi.fedoraproject.org/koji/srcpkg/ocaml-ocamlbuild. I'm not sure from where the 0.10.0 came from :-/

I see there is definitely bug somewhere, but it looks like this is on the-new-hotness side

**Comment by jamesjer** [comment]
Right, and in my case the email said the old version of pynac was 0.4.2, but it was actually 0.7.27.

As for the change to the wrong project, I was hoping you would have some logs that would point to who or what changed the project.  I saw that a lot of pypi.org projects were added recently, whether or not there are corresponding Fedora packages, and I wondered if whatever/whoever did that changed pynac.  If so, I would like to speak to the responsible party about not assuming that name equality implies project equality.  Do such logs exist?

**Comment by Zlopez** [comment]
The Pypi projects are added automatically, but they newer create any mapping to distros, just project itself.

Otherwise this information should be able to dig from [fedora messages](https://datagrepper-monitor-dashboard.app.os.fedoraproject.org/raw?topic=anitya.project.edit) If you really want to know who did it.

**Comment by jamesjer** [comment]
Okay ... but there is no mapping for the pypi.org pynac project, just the one I maintain.  If one should have been created automatically, it didn't work.

Clicking on that fedora messages link gives me "504 Gateway Time-out   The server didn't respond in time."  I've tried 3 times now, with the same result each time.

**Comment by Zlopez** [comment]
It only creates new projects as they are created in pypi by listening to libraries.io announcements, so if pynac isn't project that was created in last year, than it wouldn't be catched by this script.

**Comment by Zlopez** [comment]
There were plenty of fixes in the-new-hotness till this was opened, is this still happening?
